### PR TITLE
Feature/ab#20 event list

### PR DESCRIPTION
**Pull request by BerendWouters**

### CODE REVIEW

File: Singer.API/Services/EventService.cs [hunk 1]
```diff
@@ -34,5 +36,42 @@ public EventService(ApplicationDbContext context, IMapper mapper) : base(context
...
          return filterExpression;
       }
+
+      public IReadOnlyList<EventDescriptionDTO> GetPublicEvents(SearchEventParamsDTO searchEventParamsDto)
+      {
+         var today = DateTime.Today;
+
+         Expression<Func<Event, bool>> useStartDate =
+            x => x.StartDate >= today &&
+                 (!searchEventParamsDto.StartDate.HasValue || x.StartDate == searchEventParamsDto.StartDate.Value) &&
```
Comment (vyruz1986) [inline Singer.API/Services/EventService.cs:46]: 2nd part should be `x.StartDate >= searchEventParamsDto.StartDate.Value`. Currently we're only showing events on the exact date you've chosen

File: Singer.API/Profiles/EventProfile.cs [hunk 2]
```diff
@@ -25,9 +26,13 @@ public EventProfile()
...
             .ForMember(x => x.AllowedAgeGroups, opt => opt.MapFrom(src =>
                 src.AllowedAgeGroups.Sum(x => Convert.ToInt32(x))));
+
+         CreateMap<Event, EventDescriptionDTO>()
+            .ForMember(x => x.AgeGroups, opt => opt.MapFrom(src =>
+               ToAgeGroupList(src.AllowedAgeGroups)));
       }
 
-      private List<AgeGroup> ToAgeGroupList(AgeGroup bitmap)
+      public static List<AgeGroup> ToAgeGroupList(AgeGroup bitmap)
```
Comment (vyruz1986) [inline Singer.API/Profiles/EventProfile.cs:35]: Why was this changed?

File: Singer.API/Controllers/EventController.cs [hunk 3]
```diff
@@ -1,13 +1,24 @@
...
 {
    public class EventController : DataControllerBase<Event, EventDTO, CreateEventDTO, UpdateEventDTO>
    {
+      private readonly IEventService _eventService;
+
       public EventController(IEventService eventService) : base(eventService)
       {
+         _eventService = eventService;
+      }
+      [HttpPost("search")]
```
Comment (vyruz1986) [inline Singer.API/Controllers/EventController.cs:18]: Shouldn't we be using GET for searching?

File: Singer.API/ClientApp/src/app/modules/dashboard/components/home/home.component.ts [hunk 4]
```diff
@@ -1,9 +1,72 @@
...
+
+   onSearchEvent(searchEventData: SearchEventData){
+      this.getEvents(searchEventData).subscribe((res) =>
+         this.events =
+            res.map(r => new EventDescription(r.title, r.description, r.ageGroups, r.startDate, r.endDate)));
+   }
+}
+
+
+export class SearchEventDTO{
```
Comment (vyruz1986) [inline Singer.API/ClientApp/src/app/modules/dashboard/components/home/home.component.ts]: Shouldn't the DTO be defined in the `Singer.API\ClientApp\src\app\modules\core\models\singerevent.model.ts` file?

**Review by vyruz1986 — CHANGES_REQUESTED** [review]
Nice universal reusable solution with the age group chips. Also nice workaround with the datepicker by disabling the input field.
Please fix the `startDate` search and review my other comments (and change if appropriate).

File: Singer.API/Services/EventService.cs (hunk 1 above)
Comment (BerendWouters) [reply Singer.API/Services/EventService.cs:46]: Well I choose that option deliberately: with having that implementation, you would see all events coming after that one. Not sure if that's desired.

File: Singer.API/Profiles/EventProfile.cs (hunk 2 above)
Comment (BerendWouters) [reply Singer.API/Profiles/EventProfile.cs:35]: So I could reuse it in a different class.

File: Singer.API/Controllers/EventController.cs (hunk 3 above)
Comment (BerendWouters) [reply Singer.API/Controllers/EventController.cs:18]: Well, I wanted to use the body for the DTO, otherwise the datetime-stuff would become annoying again, especially when using it in the URL (and with optional search params). And the abstraction of the databaseservice would require me to do more work.
TLDR: yes, but no.

File: Singer.API/ClientApp/src/app/modules/dashboard/components/home/home.component.ts (hunk 4 above)
Comment (BerendWouters) [reply Singer.API/ClientApp/src/app/modules/dashboard/components/home/home.component.ts]: Good point.

File: Singer.API/Services/EventService.cs (hunk 1 above)
Comment (vyruz1986) [reply Singer.API/Services/EventService.cs:46]: In that case, the UX is not clear enough :) The way it looked to me I assumed the StartDate field should be a 'from' and the EndDate field a 'to' filter

File: Singer.API/Controllers/EventController.cs (hunk 3 above)
Comment (vyruz1986) [reply Singer.API/Controllers/EventController.cs:18]: agree, good points 👍

File: Singer.API/Services/EventService.cs (hunk 1 above)
Comment (vyruz1986) [reply Singer.API/Services/EventService.cs:46]: Let's merge as-is and see how StG responds to it, perhaps it's just me misinterpreting the field

**Review by vyruz1986 — APPROVED** [review]

File: Singer.API/Services/EventService.cs (hunk 1 above)
Comment (BerendWouters) [reply Singer.API/Services/EventService.cs:46]: No the UI is clear, because that's the point: you're interested in events starting on that date, not in the ones after that.
But this is a more functional requirement, so we'll ask the client :)
